### PR TITLE
vm_create: Role for creating VMs on target(s)

### LINKED ISSUE #51: Add a role to create a new VM
Write a role to create a new libvirt VM.

### PR DESCRIPTION
**Pull request by sbates130272**
This role can be used to create Virtual Machines (VMs) on remote target(s) and is based on [1]. Note this role depends on the qemu_setup role.

[1]: https://www.redhat.com/sysadmin/build-VM-fast-ansible

Fixes #51.

### CODE REVIEW

File: roles/vm_create/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+- name: Get VMs list
+  community.libvirt.virt:
+    command: list_vms
+  register: existing_vms
+  changed_when: false
+
+- name: Create VM if it does not exist
+  when: vm_create_vm_name not in existing_vms.list_vms
+  block:
+
```
Comment (iomartin) [inline roles/vm_create/tasks/main.yml:16]: nit: empty line

File: roles/vm_create/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    - name: Use virt-install-ubuntu in qemu-minimal
+      ansible.builtin.shell:
+        chdir: ~/Projects/qemu-minimal/libvirt
```
Comment (iomartin) [inline roles/vm_create/tasks/main.yml:19]: Maybe make this parameterized? Or does qemu-setup installs the script here?

Comment (sbates130272) [reply roles/vm_create/tasks/main.yml:19]: qemu_setup puts it here. Perhaps longer term we make the Projects folder location a parameter for all the roles that reference it but that is not part of this PR.

File: roles/qemu_setup/tasks/main.yml [hunk 3]
```diff
@@ -17,13 +17,47 @@
...
+  become: false
+
+- name: Enable libvirt_guest dns via nss
+  ansible.builtin.lineinfile:
+    path: /etc/nsswitch.conf
+    regex: ^(hosts:.*)
+    line: \1 libvirt_guest
+    backrefs: true
+  become: true
+  when: not libvirt_guest.found
```
Comment (iomartin) [inline roles/qemu_setup/tasks/main.yml:59]: I haven't tried this myself, but I feel that this and the previous step could be done in a single step. Might need a bit of playing with the regexes.

Comment (sbates130272) [reply roles/qemu_setup/tasks/main.yml:59]: I looked at that @iomartin and it is not easy and makes the code quite hard to follow. I think this is a cleaner approach.

**Review by iomartin — APPROVED** [review]
